src/modules/esl-drag-to-scroll/core/esl-drag-to-scroll.mixin.ts

```diff
@@ -0,0 +1,141 @@
+  }
+
+  /** Merged configuration object combining default and raw configuration */
+  @memoize()
+  public get config(): ESLDragToScrollConfig {
+    return {...ESLDragToScrollMixin.DEFAULT_CONFIG, ...this.rawConfig};
+  }
+
+  /** Flag indicating whether text is selected */
+  public get hasSelection(): boolean {
+    const selection = document.getSelection();
+    if (!selection || !this.config.selection) return false;
+    // Prevents draggable state if the text is selected
+    return !selection.isCollapsed && this.$host.contains(selection.anchorNode);
+  }
+
+  protected override attributeChangedCallback(name: string, oldValue: string, newValue: string): void {
+    memoize.clear(this, 'config');
+  }
+
+  /** @returns the offset of the pointer event relative to the start event */
+  public getEventOffset(event: PointerEvent): Point {
+    if (!this.startEvent || event.type === 'pointercancel') return {x: 0, y: 0};
+    const x = event.clientX - this.startEvent.clientX;
+    const y = event.clientY - this.startEvent.clientY;
+    return {x, y};
+  }
+
+  /** Scrolls the host element by the specified offset */
+  public scrollBy(offset: Point): void {
+    if (this.config.axis === 'x' || this.config.axis === 'both') {
+      this.$host.scrollLeft = this.startScrollLeft - offset.x;
+    }
+
+    if (this.config.axis === 'y' || this.config.axis === 'both') {
+      this.$host.scrollTop = this.startScrollTop - offset.y;
+    }
+  }
+
+  /** Handles the pointerdown event to start dragging */
+  @listen('pointerdown')
+  private onPointerDown(event: PointerEvent): void {
+    this.startEvent = event;
+    this.startScrollLeft = this.$host.scrollLeft;
+    this.startScrollTop = this.$host.scrollTop;
+
+    this.$$on({group: 'pointer'});
+  }
+
+  /** Handles the pointermove event to perform scrolling while dragging */
+  @listen({auto: false, event: 'pointermove', group: 'pointer'})
+  private onPointerMove(event: PointerEvent): void {
+    const offset = this.getEventOffset(event);
+
+    if (!this.isDragging) {
+      // Stop tracking if there was a selection before dragging started
+      if (this.hasSelection) return this.onPointerUp(event);
+      // Does not start dragging mode if offset have not reached tolerance
+      if (Math.abs(Math.max(Math.abs(offset.x), Math.abs(offset.y))) < this.config.tolerance) return;
+      this.isDragging = true;
+    }
+
+    this.$host.setPointerCapture(event.pointerId);
+    this.scrollBy(offset);
+  }
+
+  /** Handles the pointerup and pointercancel events to stop dragging */
+  @listen({auto: false, event: 'pointerup pointercancel', group: 'pointer'})
+  private onPointerUp(event: PointerEvent): void {
+    if (!this.isDragging) return;
+
+    this.isDragging = false;
+    this.$$off({group: 'pointer'});
+
+    if (this.$host.hasPointerCapture(event.pointerId)) {
+      this.$host.releasePointerCapture(event.pointerId);
+    }
+
+    this.scrollBy(this.getEventOffset(event));
```

Comment:
To avoid glitching after single drug reject (appers due to onPointerUp called from pointer move when isDrugging is not active, so the call doesn't ensure subscription), I'd recommend the following order of calls:
```suggestion
    this.$$off({group: 'pointer'});
    if (this.$host.hasPointerCapture(event.pointerId)) {
      this.$host.releasePointerCapture(event.pointerId);
    }

    if (this.isDragging) this.scrollBy(this.getEventOffset(event));
    this.isDragging = false;
```